### PR TITLE
Fluent test framework

### PR DESCRIPTION
**Pull request by gortiz**
I found very difficult to understand most of the tests Pinot has. Some of them need to be complex given they are testing different service topologies or casuistics, but sometimes we just want to test a simple operator and in order to do so we need to create new classes that derive from BaseQueriesTest, which are usually not easy to read. Specially when the data the test will use is defined in csvs or other files.

This PR adds a fluent framework on top of BaseQueriesTest so we can define different tests in the same class in a fluent way that are close to the usual [When-Given-Then](https://martinfowler.com/bliki/GivenWhenThen.html) tests.

This PR includes an example in pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunctionTest.java, although I think we can improve that test before merging this PR.

I think a framework like would be easier to read and I plan to use it to cover most of the aggregation operations in order to feel confident once I modify them to support `nullHandlingEnabled`.

### CODE REVIEW

File: pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/AbstractAggregationFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.utils.builder.TableConfigBuilder;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+
+public class AbstractAggregationFunctionTest {
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/AbstractAggregationFunctionTest.java:44]:
```suggestion
public abstract class BaseAggregationFunctionTest {
```

File: pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunctionTest.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+  }
+
+  @AfterClass
+  void destroyBaseDir()
+      throws IOException {
+    if (_baseDir != null) {
+      FileUtils.deleteDirectory(_baseDir);
+    }
+  }
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunctionTest.java:53]: Do we need this part? It is the same as the base class

**Review by Jackie-Jiang — APPROVED** [review]

File: pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/AbstractAggregationFunctionTest.java (hunk 1 above)
Comment (gortiz) [reply pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/AbstractAggregationFunctionTest.java:44]: Couldn't we just use abstract? We use that name pattern in several places (`AbstractMetric`, `AbstractRule`, `AbstractCommand`, `AbstractPlanNode`...) and it is also a common name in JDK (`AbstractList`, `AbstractChronology`...)

What I'm actually going to add for sure is the abstract class qualifier.

File: pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunctionTest.java (hunk 2 above)
Comment (gortiz) [reply pinot-core/src/test/java/org/apache/pinot/core/query/aggregation/function/CountAggregationFunctionTest.java:53]: Removed

**Pull request merged**